components/ws-proxy/pkg/sshproxy/server.go

```diff
@@ -484,6 +497,58 @@ func (s *Server) GetWorkspaceSSHKey(ctx context.Context, workspaceIP string, sup
 	return key, userName, nil
 }
 
+func (s *Server) GenerateSSHCert(ctx context.Context, userName string) (ssh.Signer, error) {
+	// prepare certificate for signing
+	nonce := make([]byte, 32)
+	if _, err := io.ReadFull(rand.Reader, nonce); err != nil {
+		return nil, xerrors.Errorf("failed to generate signed SSH key: error generating random nonce: %w", err)
+	}
+
+	pk, pv, err := ed25519.GenerateKey(rand.Reader)
+	if err != nil {
+		return nil, err
+	}
+	b, err := ssh.NewPublicKey(pk)
+	if err != nil {
+		return nil, err
+	}
+
+	priv, err := ssh.NewSignerFromSigner(pv)
+	if err != nil {
+		return nil, err
+	}
+
+	now := time.Now()
+
+	certificate := &ssh.Certificate{
+		Serial:          0,
+		Key:             b,
+		KeyId:           "ws-proxy",
+		ValidPrincipals: []string{userName},
+		ValidAfter:      uint64(now.Add(time.Minute * -10).In(time.UTC).Unix()),
+		ValidBefore:     uint64(now.Add(time.Minute * 10).In(time.UTC).Unix()),
```

Comment:
Do I understand this correctly that the cert here is only valid for 10min because it's just used for the single SSH login operation?

```suggestion
		ValidAfter:      uint64(now.Add(-10 * time.Minute).In(time.UTC).Unix()),
		ValidBefore:     uint64(now.Add(10 * time.Minute).In(time.UTC).Unix()),
```